### PR TITLE
[View Payments] Replacing usage of moment with date-fns

### PR DESCRIPTION
**Pull request by jerekshoe**
## Summary
Replacing usage of `moment.js` with `date-fns` in the application. Usage of `moment.js` was found to be isolated to 1 file in this application (`src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js`). Added functions for determining if a date string produces a valid date and for formatting a date string to a specific format

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#79738

## Testing done
Added tests for the `formateDate` and `isValidDate` helpers

## Screenshots
**Note:** The table is not sorted, it displays the payments in whichever order they are returned in from vets-api
Screenshot 2024-04-02 at 10 26 39 AM

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+const parseFormatStr = 'MM/dd/yyyy';
+
+export const isValidDate = dateString => {
+  const parsedDate = parse(dateString, parseFormatStr, new Date());
+
+  return isValid(parsedDate);
+};
+
+export const formatDate = dateString => {
+  const formatString = 'MMMM d, yyyy';
```
Comment (pmclaren19) [inline src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:13]: should this be ''MMMM dd, yyyy' ?

Comment (pmclaren19) [reply src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:13]: I see references of both but more of the all caps version...not sure if it matters though?

Comment (jerekshoe) [reply src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:13]: No, if you capitalize the d and y they both have different meanings. [Docs](https://date-fns.org/v3.6.0/docs/format)

File: src/applications/disability-benefits/view-payments/README.md [hunk 2]
```diff
@@ -46,187 +46,186 @@ When the front end calls [/profile/payment_history](https://github.com/departmen
     replacing the `payment_history` method in `/app/services/bgs/payment_service.rb` with the following:
 
     ```ruby
-    def payment_history(person)
-        response = {
-            :payments => {
-                :payment => [
+    def payment_history
```
Comment (jerekshoe) [inline src/applications/disability-benefits/view-payments/README.md:49]: The formatting changes are a result of the changes made by the formatter in vets-api, I copied the code from there after I had pasted it from here and let the formatter do its thing

**Review by pmclaren19 — APPROVED** [review]

**Pull request merged**